_posts/2024-12-19-moving-data-between-JBOD-disks-using-cruise-control.md

```diff
@@ -0,0 +1,345 @@
+---
+layout: post
+title: "Moving data between the JBOD disks using Cruise Control"
+date: 2024-12-19
+author: shubham_rawat
+---
+
+Apache Kafka is a platform which provides durability and fault tolerance by storing messages on disks and JBOD storage is one of the storage configuration types supported by Kafka.
+The JBOD data storage configuration allows Kafka brokers to make use of multiple disks.
+Using JBOD storage, you can increase the data storage capacity for Kafka nodes, which can further lead to performance improvements.
+In case you plan to remove a disk, and it contains some partition replicas, then you need to make sure that data is safely moved to some other disks.
+If the data is not removed from the disk, and it is removed then potential data loss can happen.
+Currently, moving data between the JBOD disks is done using the `kafka-reassign-partitions.sh` tool which is not very user-friendly, therefore in Strimzi 0.45.0 we are introducing the ability to move data between the JBOD disks using Cruise Control
+
+## Cruise Control to move data between JBOD disks
+
+This feature will allow you to move the data between the JBOD disks using the `KafkaRebalance` custom resource that we have in Strimzi.
+This feature makes use of the `remove-disks` endpoint of Cruise Control that triggers a rebalancing operation which moves replicas, starting with the largest and proceeding to the smallest, to the remaining disks. 
+
+## Setting up the environment
+
+Let's set up a cluster to work through an example demonstrating this feature.
+To get the Kafka cluster up and running, we will first have to install the Strimzi Cluster Operator and then deploy the `Kafka` resource.
+You can install the Cluster Operator with any installation method you prefer.
+For example by following the [Stimzi Quickstart Guide](https://strimzi.io/quickstarts/).
```

Comment:
Should you point to the docs instead? The quickstart deploys a Kafka cluster as well.